### ISSUE TITLE
lexical_cast for integers doesn't check if the string is fully consumed

### ISSUE THREAD

**Issue by SX91** [CLIUtils/CLI11#67]
CLI11 doesn't check for invalid strings like `100500blabla` and happily consumes it.

I guess `lexical_cast` for integers should look like this:
```C++
/// Integers / enums
template <typename T,
          enable_if_t<std::is_integral<T>::value || std::is_enum<T>::value, detail::enabler> = detail::dummy>
bool lexical_cast(std::string input, T &output) {
    try {
        size_t n = 0;
        output = static_cast<T>(std::stoll(input, &n, 0)); // note: this way conversion works for HEX too
        return (n == input.size());
    } catch(const std::invalid_argument &) {
        return false;
    } catch(const std::out_of_range &) {
        return false;
    }
}

/// Floats
template <typename T, enable_if_t<std::is_floating_point<T>::value, detail::enabler> = detail::dummy>
bool lexical_cast(std::string input, T &output) {
    try {
        size_t n = 0;
        output = static_cast<T>(std::stold(input, &n));
        return n == input.size();
    } catch(const std::invalid_argument &) {
        return false;
    } catch(const std::out_of_range &) {
        return false;
    }
}
```

**Comment by henryiii** [comment]
Could you make a PR? Looks like a good change...

**Comment by henryiii** [comment]
Closed with #68
